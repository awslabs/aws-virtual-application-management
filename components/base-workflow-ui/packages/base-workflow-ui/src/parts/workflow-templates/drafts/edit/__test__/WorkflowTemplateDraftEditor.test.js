/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowTemplateDraftEditor from '../WorkflowTemplateDraftEditor';

describe('WorkflowTemplateDraftEditor', () => {
  it('WorkflowTemplateDraftEditor should renders correctly', () => {
    const workflowTemplateDraftsStore = {
      ready: 'store-ready',
      load: jest.fn(),
      startHeartbeat: jest.fn(),
      list: [],
      getDraft: jest.fn(() => {
        return { template: {} };
      }),
      hasDraft: jest.fn(),
    };
    const stepTemplatesStore = {
      ready: 'store-ready',
      load: jest.fn(),
      startHeartbeat: jest.fn(),
      list: [],
      getDraft: jest.fn(() => {
        return { template: {} };
      }),
      hasDraft: jest.fn(),
    };
    const className = 'test-className';
    const match = { params: { test: 'test' } };
    const userDisplayName = 'test-userDisplayName';
    expect(
      renderer
        .create(
          <Provider workflowTemplateDraftsStore={workflowTemplateDraftsStore} stepTemplatesStore={stepTemplatesStore}>
            <WorkflowTemplateDraftEditor.WrappedComponent
              workflowTemplateDraftsStore={workflowTemplateDraftsStore}
              stepTemplatesStore={stepTemplatesStore}
              className={className}
              match={match}
              userDisplayName={userDisplayName}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
