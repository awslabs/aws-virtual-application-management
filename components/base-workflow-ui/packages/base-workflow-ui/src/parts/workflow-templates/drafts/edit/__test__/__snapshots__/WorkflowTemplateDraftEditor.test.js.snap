// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowTemplateDraftEditor WorkflowTemplateDraftEditor should renders correctly 1`] = `
<div
  className="ui container mt3"
>
  <div
    className="ui breadcrumb block mb3"
  >
    <a
      className="section"
      onClick={[Function]}
    >
      Workflow Template Drafts
    </a>
    <i
      aria-hidden="true"
      className="right angle icon divider"
      onClick={[Function]}
    />
    <div
      className="active section"
      onClick={[Function]}
    >
      undefined
    </div>
  </div>
  <div
    className="p0"
  >
    <div
      className="ui negative message clearfix"
    >
      <div
        className="header"
      >
        A problem was encountered
      </div>
      <p>
        The workflow template draft "undefined" is not available
      </p>
    </div>
  </div>
</div>
`;
